packages/playwright/src/transform/compilationCache.ts

```diff
@@ -168,9 +172,22 @@ export function addToCompilationCache(payload: SerializedCompilationCache) {
   }
 }
 
-function calculateCachePath(filePath: string, hash: string): string {
+function calculateCacheFolderName(filePath: string): string {
+  const pathHash = crypto.createHash('sha1').update(filePath).digest('hex');
+  return pathHash.substring(0, 7);
+}
+
+function calculateCachePath(filePath: string, folderName: string, hash: string): string {
   const fileName = path.basename(filePath, path.extname(filePath)).replace(/\W/g, '') + '_' + hash;
-  return path.join(cacheDir, hash[0] + hash[1], fileName);
+  return path.join(cacheDir, folderName, fileName);
```

Comment:
`chacheDir/${fileNameHash[0]}${fileNameHash[1]}/${baseName}_${fileHash}_${contentHash}`

Trim the content hash as well.